ctest fails on the target, because it cannot find CMake.cmake inside
CMAKE_DATA_DIR (typically set to "/usr/share/cmake-3.0"): all *.cmake
files are removed at build time via the target-finalize rule.

This buildroot-specific patch makes sure ctest looks also for
"Modules/CMake.cmake.ctest" before complaining

[Vincent: tweak patch for 3.6.3]

Signed-off-by: Davide Viti <zinosat@tiscali.it>
Signed-off-by: Vicente Olivert Riera <Vincent.Riera@imgtec.com>

diff -rup a/Source/cmake.cxx b/Source/cmake.cxx
--- a/Source/cmake.cxx	2016-07-07 15:47:27.000000000 +0100
+++ b/Source/cmake.cxx	2016-07-14 10:14:59.914265515 +0100
@@ -771,7 +771,12 @@ int cmake::AddCMakePaths()
                       "Path to cpack program executable.", cmState::INTERNAL);
 #endif
   if (!cmSystemTools::FileExists(
-        (cmSystemTools::GetCMakeRoot() + "/Modules/CMake.cmake").c_str())) {
+        (cmSystemTools::GetCMakeRoot() + "/Modules/CMake.cmake").c_str()) &&
+	!cmSystemTools::FileExists(
+	  (cmSystemTools::GetCMakeRoot()+"/Modules/CMake.cmake.ctest").c_str()
+        )
+     )
+  {
     // couldn't find modules
     cmSystemTools::Error(
       "Could not find CMAKE_ROOT !!!\n"
